### PR TITLE
[wasm] Update javascript API usage in samples and tests

### PR DESCRIPTION
**Pull request by maraf**
- Use `createDotnetRuntime` with object instead of callback where possible.
- Replace `bind_static_method` with `getAssemblyExports`.

### CODE REVIEW

File: src/mono/sample/mbr/browser/Program_v1.cs [hunk 1]
```diff
@@ -4,29 +4,32 @@
...
-	static DeltaHelper replacer = DeltaHelper.Make ();
+        static DeltaHelper replacer = DeltaHelper.Make ();
 
         public static void Main(string[] args)
         {
             Console.WriteLine ("Hello, World!");
         }
 
+        [JSExport]
         [MethodImpl(MethodImplOptions.NoInlining)]
```
Comment (pavelsavara) [inline src/mono/sample/mbr/browser/Program_v1.cs:22]: I think you could remove all `MethodImplOptions.NoInlining` where you have `[JSExport]`

File: src/mono/sample/wasm/browser-eventpipe/main.js [hunk 2]
```diff
@@ -85,25 +85,18 @@ function getOnClickHandler(startWork, stopWork, getIterationsDone) {
 }
 
 async function main() {
-    const { MONO, BINDING, Module, RuntimeBuildInfo } = await createDotnetRuntime(() => {
-        return {
-            disableDotnet6Compatibility: true,
-            configSrc: "./mono-config.json",
-        }
+    const { MONO, Module, getAssemblyExports } = await createDotnetRuntime({
+        disableDotnet6Compatibility: true,
```
Comment (pavelsavara) [inline src/mono/sample/wasm/browser-eventpipe/main.js:89]: `disableDotnet6Compatibility` is true by default.

File: src/mono/sample/wasm/console-node-es6/main.mjs [hunk 3]
```diff
@@ -8,10 +8,10 @@ import process from 'process'
  */
 const createDotnetRuntimeTyped = createDotnetRuntime;
 
-const { runMainAndExit } = await createDotnetRuntimeTyped(() => ({
+const { runMainAndExit } = await createDotnetRuntimeTyped({
```
Comment (pavelsavara) [inline src/mono/sample/wasm/console-node-es6/main.mjs:11]: the folder could be renamed to `console-node` and same for `v8-node`

File: src/mono/wasm/debugger/tests/debugger-test/debugger-driver.html [hunk 4]
```diff
@@ -6,15 +6,16 @@
...
-				this.int_add = App.runtime.BINDING.bind_static_method("[debugger-test] Math:IntAdd");
-				this.use_complex = App.runtime.BINDING.bind_static_method("[debugger-test] Math:UseComplex");
-				this.delegates_test = App.runtime.BINDING.bind_static_method("[debugger-test] Math:DelegatesTest");
-				this.generic_types_test = App.runtime.BINDING.bind_static_method("[debugger-test] Math:GenericTypesTest");
-				this.outer_method = App.runtime.BINDING.bind_static_method("[debugger-test] Math:OuterMethod");
-				this.async_method = App.runtime.BINDING.bind_static_method("[debugger-test] Math/NestedInMath:AsyncTest");
-				this.method_with_structs = App.runtime.BINDING.bind_static_method("[debugger-test] DebuggerTests.ValueTypesTest:MethodWithLocalStructs");
-				this.run_all = App.runtime.BINDING.bind_static_method("[debugger-test] DebuggerTest:run_all");
+			init: async () => {
+				const exports = await App.runtime.getAssemblyExports("debugger-test.dll");
```
Comment (pavelsavara) [inline src/mono/wasm/debugger/tests/debugger-test/debugger-driver.html:10]: I love this change

**Comment by maraf** [comment]
The `debugger-driver.html` still has a `bind_static_method` for calls that pass method from C# as part of the tests. I'll address these in a separate PR.

File: src/mono/wasm/debugger/tests/debugger-test/debugger-main.js [hunk 5]
```diff
@@ -22,7 +22,7 @@ try {
         },
     }));
     App.runtime = runtime;
-    App.init()
+    await App.init();
```
Comment (pavelsavara) [inline src/mono/wasm/debugger/tests/debugger-test/debugger-main.js:25]: @radical FYI app will be ready only after this is resolved

**Review by pavelsavara — APPROVED** [review]
LGTM

File: src/mono/wasm/debugger/tests/debugger-test/debugger-main.js (hunk 5 above)
Comment (pavelsavara) [reply src/mono/wasm/debugger/tests/debugger-test/debugger-main.js:25]: Related to https://github.com/dotnet/runtime/issues/73528

Comment (pavelsavara) [reply src/mono/wasm/debugger/tests/debugger-test/debugger-main.js:25]: I think @thaystg  wanted to move `mono_wasm_runtime_ready` console log line into the unit test. That would be right place and right signal

**Comment by maraf** [comment]
One debugger test timed out on `runtime-staging`, but passed on `runtime-wasm`.

**Pull request merged**